src/aleph_client/commands/program.py

```diff
@@ -83,8 +84,40 @@ def upload(
         help="Hash of the runtime to use for your program. Defaults to aleph debian with Python3.8 and node. You can also create your own runtime and pin it",
     ),
     beta: bool = typer.Option(False),
+
+
+
     debug: bool = False,
     persistent: bool = False,
+    persistent_volume: Optional[List[str]] = typer.Option(
+        None,
+        help= '''Takes 3 parameters                                                                                                                             
+        A persistent volume is allocated on the host machine at any time                                             
+        eg:                                                                   
+        --persistent_volume  host,my-volume,100
+        use , to seperate the parameters'''),
+
+    ephemeral_volume: Optional[List[str]] = typer.Option(
+        None,
+        help=
+        '''Takes 1 parameter Only                                           
+        Ephemeral volumes can move and be removed by the host,Garbage collected basically, when the VM isn't running                                  
+        eg:                                                                     
+         --ephemeral_volume 100 '''),
+
+    immutable_volume: Optional[List[str]] = typer.Option(
+        None,
+        help=
+        '''Takes 3 parameters                                           
+         Immutable volume is one whose contents do not change                                   
+         eg: Use , to seperate the parameters                                                                      
+        --immutable_volume 25a393222692c2f73489dc6710ae87605a96742ceef7b91de4d7ec34bb688d94,True,/mnt/volume
```

Comment:
The example string does not seem to line up with how `volume_to_dict` is defined:

```py
def volume_to_dict(volume: List[str]) -> Optional[Dict[str, str]]:
    if not volume:
        return None
    dict_store = {}
    for word in volume:
        split_values = word.split(",")
        for param in split_values:
            p = param.split("=")  # we split on "=" here
 ```
 So the help string should follow the convention as laid out in the linked issue: https://github.com/aleph-im/aleph-client/issues/116#issuecomment-1298863171
 using commas instead of semicolons, as we discussed.

Reply:
@MHHukiewitz need the discussion on that

Reply:
Fixed